packages/mintlify/src/local-preview/index.ts

```diff
@@ -52,77 +54,103 @@ const dev = async () => {
-  const currBranch = shellExec("git rev-parse --abbrev-ref HEAD").stdout;
-  if (
-    gitPullStatus.startsWith(
-      "commit 78d764335877932a0dc305d615411dedd18bd18a"
-    ) ||
-    currBranch === "legacy-components-import"
-  ) {
-    await fse.emptyDir(MINT_PATH);
-  }
-
-  let runYarn = true;
-  const gitInstalled = shell.which("git");
-  let firstInstallation = false;
-  const gitRepoInitialized = await pathExists(
-    path.join(DOT_MINTLIFY, "mint", ".git")
-  );
-  if (!gitRepoInitialized) {
-    firstInstallation = true;
-    if (gitInstalled) {
-      logger.start("Initializing local Mintlify instance...");
-      shellExec("git init");
-      shellExec(
-        "git remote add -f mint-origin https://github.com/mintlify/mint.git"
-      );
-    } else {
-      logger.fail(
-        "git must be installed (https://github.com/git-guides/install-git)"
-      );
-      process.exit(1);
-    }
-  }
 
+  // Can only run offline if Mint was already downloaded
   const internet = await isInternetAvailable();
-  let pullOutput = null;
-  if (internet && gitInstalled) {
-    shellExec("git config core.sparseCheckout true");
-    shellExec('echo "client/" >> .git/info/sparse-checkout');
-    pullOutput = shellExec("git pull mint-origin main").stdout;
-    shellExec("git config core.sparseCheckout false");
-    shellExec("rm .git/info/sparse-checkout");
-  }
-  if (pullOutput === "Already up to date.\n") {
-    runYarn = false;
+  if (!internet && !(await pathExists(CLIENT_PATH))) {
+    logger.fail("Cannot download Mint without an internet connection.");
+    process.exit(1);
   }
-  shell.cd(CLIENT_PATH);
-  if (internet && (runYarn || !(await nodeModulesExists()))) {
-    if (firstInstallation) {
-      logger.succeed("Local Mintlify instance initialized");
+
+  // Avoid checking if we are on the target Mint if we are offline
+  if (internet) {
+    const mintVersionExists = await pathExists(VERSION_PATH);
+
+    // We always download the target version of Mintlify if the mint-version.txt file doesn't exist.
+    // We do this because users updating from an older version of the CLI never have mint-version.txt set.
+    let downloadTargetMint = !mintVersionExists;
+
+    // Download target mint if the current version is different. Keep in mind this also allows
+    // downgrading to an older version of Mintlify by installing an older CLI version.
+    if (mintVersionExists) {
+      const currVersion = fse.readFileSync(VERSION_PATH, "utf8");
+      if (currVersion !== TARGET_MINT_VERSION) {
+        downloadTargetMint = true;
+      }
     }
-    logger.start("Updating dependencies...");
-    ensureYarn(logger);
-    shellExec("yarn");
-    if (firstInstallation) {
-      logger.succeed("Installation complete");
-    } else {
-      logger.succeed("Dependencies updated");
+
+    if (downloadTargetMint) {
+      // Delete any existing contents
+      fse.emptyDirSync(MINT_PATH);
+
+      logger.text = "Downloading Mint...";
```

Comment:
The goal of the CLI isn't to let people know about the existence of mint. In fact we are purposefully trying to make mint accessible in a way in which you don't have to know about it.
```suggestion
      logger.text = "Downloading packages...";
```